### PR TITLE
[N/A] Update Deployed File Structure + Add Configs

### PR DESCRIPTION
**Pull request by rypete**
Updates the deployment process to deploy the contents of the `dist` directory only.  Also adds in `.eslintrc` and `tsconfig-services` to allow projects to extend from.

### CODE REVIEW

File: src/webpack/webpackTools.ts [hunk 1]
```diff
@@ -13,7 +13,8 @@ export interface CustomWebpackOptions extends webpack.Options.Optimization {
      */
     minify?: boolean;
     /**
-     * If the resulting module should inject itself into the window object to make itself easily accessible within HTML.  Defaults to false.
+     * If the resulting module should inject itself into the window object to make itself
+     * easily accessible within HTML.  Defaults to false.
```
Comment (bryangaleopenfin) [inline src/webpack/webpackTools.ts]: Double space

File: src/cli.ts [hunk 2]
```diff
@@ -59,7 +75,7 @@ function startCommandProcess(args: CLIArguments) {
 }
 
 /**
- * Initiates a webpack build for the extending project
+ * Initiates a webpack build for the extending project.
```
Comment (bryangaleopenfin) [inline src/cli.ts]: What's the standard. This makes it inconsistent with other comments in this file

**Review by bryangaleopenfin — APPROVED** [review]
Approved with minor comments

File: package.json [hunk 3]
```diff
@@ -1,22 +1,18 @@
...
-    "prepack": "npm run clean && npm run build",
+    "check": "eslint src demo --ext .ts --ext .tsx --config src/config/.eslintrc.json",
+    "clean": "rimraf dist",
+    "fix": "eslint src demo --ext .ts --ext .tsx --config src/config/.eslintrc.json --fix",
+    "build": "npm run clean && tsc -p tsconfig.json && node src/fileCopy.js",
     "demo": "npm run build && cd demo && npm run start",
     "demo:build": "npm run build && cd demo && npm run build",
-    "start": "npm run build && node dist/cli.js start"
+    "start": "npm run build && node dist/cli.js start",
+    "deploy:test": "npm run build && cd dist && npm pack"
```
Comment (pjbroadbent) [inline package.json]: Maybe name this `demo:pack`, to align with the `demo` and `demo:build` scripts?..

It doesn't "deploy" anything, and "test" in a script kinda implies something more like unit testing (to me, at least).

File: package.json [hunk 4]
```diff
@@ -38,18 +34,20 @@
...
     "url-loader": "^1.0.1",
     "webpack": "^4.16.1",
     "webpack-cli": "^3.0.8",
-    "webpack-dev-middleware": "^3.6.0"
+    "webpack-dev-middleware": "^3.6.0",
+    "eslint": "^5.15.3",
+    "eslint-plugin-import": "^2.16.0",
+    "eslint-plugin-react": "^7.12.4",
+    "@typescript-eslint/eslint-plugin": "^1.5.0",
+    "@typescript-eslint/parser": "^1.5.0"
```
Comment (pjbroadbent) [inline package.json]: Can we keep this list sorted, to prevent unnecessary changes/conflicts when next adding or removing a package?

You can [apparently](https://stackoverflow.com/a/45187142) get NPM to do this for you - just run `npm remove -S example; npm remove -D example`

**Comment by bryangaleopenfin** [comment]
Couple of general things that i'm fine with being dismissed as outside this PR, but think aree worth raising:

- Is there a problem in server.js? These lines:
```
const fetchRequest = await fetch.default(getProviderUrl(args.providerVersion)).catch((err: string) => {
    throw new Error(err);
});
```

Should they be outside the `!args.noDemo` check?

- This message in webpackTools.ts:

`'Using 'npm run build' (or build:dev) when running locally. Can debug without building first by running 'npm start'.'`

is inaccurate now, and I'm not sure service-tooling should be making assumptions about how it's being used in a package.json file

**Comment by rypete** [comment]
@bryangaleOF 
First bullet: We discussed offline.
Second bullet:  This was left over from first gen server.js.  I have removed it now.

File: package.json (hunk 3 above)
Comment (rypete) [reply package.json]: This was a local test - I have removed it now.

File: package.json (hunk 4 above)
Comment (rypete) [reply package.json]: Fixed

**Review by pjbroadbent — APPROVED** [review]
